sycl/include/sycl/builtins.hpp

```diff
@@ -789,6 +896,79 @@ detail::enable_if_t<detail::is_svgenfloat<T>::value, T> sign(T x) __NOEXC {
   return __sycl_std::__invoke_sign<T>(x);
 }
 
+// marray common functions
+
+// TODO: can be optimized in the way math functions are optimized (usage of
+// vec<T, 2>)
+#define __SYCL_MARRAY_COMMON_FUNCTION_UNOP_OVERLOAD(NAME, ARGTYPE, ARGVAL,     \
+                                                    ITEM)                      \
+  template <typename T,                                                        \
+            typename = std::enable_if_t<detail::is_mgenfloat<T>::value>>       \
+  sycl::marray<detail::marray_element_type<T>, T::size()> NAME(ARGTYPE ARGVAL) \
```

Comment:
```suggestion
  T NAME(ARGTYPE ARGVAL) \
```

I suppose that we always return the same `marray` as we accept, right? If so, we could simplify the return type here